src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1183,14 +1183,18 @@ private void ParseModifiers(SyntaxListBuilder tokens)
                 {
                     case DeclarationModifiers.Partial:
                         var nextToken = PeekToken(1);
-                        var isPartialType = this.IsPartialType();
-                        var isPartialMember = this.IsPartialMember();
-                        if (isPartialType || isPartialMember)
+                        var isPartialFollowedByType = this.IsPartialType();
```

Comment:
I think we should change or remove the `IsPartialType` helper. It no longer does what it says it does.

Reply:
We should also add tests for the case described in the comment for `ParseIdentifierToken`